pkg/podimpersonation/podimpersonation.go

```diff
@@ -348,7 +348,19 @@ func (s *PodImpersonation) createPod(ctx context.Context, user user.Info, role *
 	if err != nil {
 		return nil, err
 	}
-
+	if _, ok := tokenSecret.Data["token"]; !ok {
+		for {
+			logrus.Debugf("wait for svc account secret to be populated with token %s", tokenSecret.Name)
+			time.Sleep(2 * time.Second)
```

Comment:
Also wondering if we should keep the wait interval configurable across rancher - but something for later

Reply:
yep maybe we could for rancher controllers, not sure if passing to steve would be easily possible

Reply:
If we could make the wait interval configure via an opt-in environment variable without adding the requirement for additional cycles, I believe that would be a great value add. That way, we take a "don't touch this unless you are told to touch this" approach